compiler/src/iree/compiler/Codegen/Common/LowerUKernelDescriptors.cpp

```diff
@@ -113,11 +108,19 @@ static LogicalResult convertToUKernelGeneric(RewriterBase &rewriter,
       }
     }
   }
+
   rewriter.setInsertionPoint(op);
-  rewriter.replaceOpWithNewOp<IREE::Codegen::UKernelGenericOp>(
-      op, op->getResults().getTypes(), name, tensorInputs, tensorOutputs,
-      otherOperands, DictionaryAttr(),
-      /*strided_outer_dims=*/0);
+  if (!provider) {
```

Comment:
We will almost always have a provider but the provider won't necessarily implement the replacement method. The comment I added below should fix that (make the existing implementation the interface default and let people override it).

Reply:
Marking resolved as part of the [previous review thread](https://github.com/iree-org/iree/pull/21681#discussion_r2302030230).